resolvers.py

```diff
@@ -0,0 +1,1979 @@
+
+    # Adding Project Data in Account Level with their respective allocations, also bulge_ratio if it is present
+    def process_account_data(self, account_tuple, account_data):
+        account_allocations = None
+        account_bulge_ratio = None
+        account_duration = account_data.get("duration")
+        account_start_date = account_data.get("start_date")
+        account_end_date = account_data.get("end_date")
+        account_category = account_data.get("category")
+        account_billability = account_data.get("billability")
+        account_projects = []
+        for project_name, project_data in account_data["children"].items():
+            project_data_processed = self.process_project_data(
+                project_name, project_data)
+            account_projects.append(project_data_processed)
+            if project_name == "Total":
+                account_allocations = project_data_processed["allocations"]
+                account_bulge_ratio = project_data_processed.get(
+                    "bulge_ratio")  # Retrieve bulge_ratio if available
+        if account_allocations is None:
+            account_allocations = account_data.get("allocations")
+        if account_bulge_ratio is None:
+            return {"account_name": account_tuple[1], "children": account_projects, "duration": account_duration, "start_date": account_start_date, "end_date": account_end_date, "category": account_category, "billability": account_billability, "allocations": account_allocations}
+        else:
+            return {"account_name": account_tuple[1], "children": account_projects, "duration": account_duration, "start_date": account_start_date, "end_date": account_end_date, "category": account_category, "billability": account_billability, "allocations": account_allocations, "bulge_ratio": account_bulge_ratio}
+
+
+class NestingOfAccountWiseHeadcount(NestingHelper):
+    @custom_resolver_check_perm
+    def __call__(self, source, info, **kwargs):
+
+        # Adding Master Filter for billability and Multiple Account Names
+        billability = kwargs.get("billability")
+        filters = kwargs.get("filters")
+        flat_data = models.AccountWiseHeadcount.objects.all()
+
+        if billability:
+            flat_data = flat_data.filter(billability__in=billability)
+
+        if filters:
+            flat_data = flat_data.filter(account_name__in=filters)
+
+        # Making a Intermediate Grouped Data for our required Nested API
+        grouped_data = self.group_data(flat_data)
+
+        # Nested API
+        nested_data = []
+        for account_tuple, account_data in grouped_data.items():
+            nested_data.append(self.process_account_data(
+                account_tuple, account_data))
+
+        return nested_data
+
+
+class GetPodDetailsResolver():
+    @custom_resolver_check_perm
+    def __call__(self, source, info, **kwargs):
+        pod_id = kwargs.get("id")
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

You are using the `get` method without a default argument to return the value of a key in a dictionary. We recommended that you use a default argument so that if the value for your key is not found, a default value is returned. If a default value is not provided and the key is not found, then `None` is returned.

[Learn more](https://docs.python.org/3/library/stdtypes.html#dict.get)